contrib/go-chi/chi/chi.go

```diff
@@ -0,0 +1,36 @@
+// Package chi provides tracing functions for tracing the go-chi/chi package (https://github.com/go-chi/chi).
+package chi // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/go-chi/chi"
+
+import (
+	"net/http"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/httputil"
+
+	"github.com/go-chi/chi"
+)
+
+// Router embeds a chi.Router, augmenting it with instrumentation. Use NewRouter to instantiate.
+type Router struct {
+	chi.Router
+	config *routerConfig
+}
+
+// NewRouter creates a new Router using the given options.
+func NewRouter(opts ...RouterOption) *Router {
+	cfg := new(routerConfig)
+	defaults(cfg)
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	return &Router{
+		Router: chi.NewRouter(),
+		config: cfg,
+	}
+}
+
+// ServeHTTP implements http.Handler, overriding the embedded chi.Router.
+func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	resource := req.Method + " " + req.URL.Path
```

Comment:
We can not use `req.URL.Path` because it could contain sensitive information such as API tokens or authentication. We would have to use something similar to what we have in the `gorilla/mux` integration or in `net/http`. Example: https://github.com/DataDog/dd-trace-go/blob/v1/contrib/net/http/http.go#L36.

Reply:
Maybe `go-chi/chi` has something similar for extracting the route pattern out of the request. Gorilla/mux has something like this: https://github.com/DataDog/dd-trace-go/blob/v1/contrib/gorilla/mux/mux.go#L92

Reply:
I think this is what you need to use: https://godoc.org/github.com/go-chi/chi#RouteContext

From that, you can use the method [`RoutePattern`](https://godoc.org/github.com/go-chi/chi#Context.RoutePattern) to get the result.